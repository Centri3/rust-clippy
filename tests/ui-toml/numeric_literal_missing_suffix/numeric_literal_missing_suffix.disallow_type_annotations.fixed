//@run-rustfix
//@aux-build:../../ui/auxiliary/proc_macros.rs:proc-macro
//@revisions: disallow_type_annotations allow_type_annotations
//@[disallow_type_annotations] rustc-env:CLIPPY_CONF_DIR=$DIR/disallow_type_annotations
//@[allow_type_annotations] rustc-env:CLIPPY_CONF_DIR=$DIR/allow_type_annotations
#![allow(unused)]
#![warn(clippy::numeric_literal_missing_suffix)]
#![feature(custom_inner_attributes)]
#![rustfmt::skip] // Not sure why rustfmt messes up the uitest commands

#[macro_use]
extern crate proc_macros;

fn main() {
    let x = 1i32;
    let x = 1.0f64;
    let x: u8 = 1u8;
    static X1: u32 = 0u32;
    const X2: u32 = 0u32;
    // Don't lint
    external! {
        let x = 1;
        let x: u8 = 1;
        static X3: u32 = 0;
        const X4: u32 = 0;
    }
    with_span! {
        span
        let x = 1;
        let x: u8 = 1;
        static X5: u32 = 0;
        const X6: u32 = 0;
    }
}
