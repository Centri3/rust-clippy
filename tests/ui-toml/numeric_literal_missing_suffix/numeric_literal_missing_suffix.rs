//@run-rustfix
//@aux-build:../../ui/auxiliary/proc_macros.rs:proc-macro
//@revisions: disallow_type_annotations allow_type_annotations
//@[disallow_type_annotations] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/numeric_literal_missing_suffix/disallow_type_annotations
//@[allow_type_annotations] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/numeric_literal_missing_suffix/allow_type_annotations
#![allow(unused)]
#![warn(clippy::numeric_literal_missing_suffix)]
#![feature(custom_inner_attributes)]
#![rustfmt::skip] // Not sure why rustfmt messes up the uitest commands

#[macro_use]
extern crate proc_macros;

fn main() {
    let x = 1;
    let x = 1.0;
    let x: u8 = 1;
    static X1: u32 = 0;
    const X2: u32 = 0;
    // Don't lint
    external! {
        let x = 1;
        let x: u8 = 1;
        static X3: u32 = 0;
        const X4: u32 = 0;
    }
    with_span! {
        span
        let x = 1;
        let x: u8 = 1;
        static X5: u32 = 0;
        const X6: u32 = 0;
    }
}
