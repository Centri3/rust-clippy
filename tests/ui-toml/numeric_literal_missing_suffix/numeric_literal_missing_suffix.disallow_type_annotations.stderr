error: this literal is missing an explicit suffix
  --> $DIR/numeric_literal_missing_suffix.rs:15:14
   |
LL |     let x = 1;
   |              ^ help: add it: `i32`
   |
   = note: `-D clippy::numeric-literal-missing-suffix` implied by `-D warnings`

error: this literal is missing an explicit suffix
  --> $DIR/numeric_literal_missing_suffix.rs:16:16
   |
LL |     let x = 1.0;
   |                ^ help: add it: `f64`

error: this literal is missing an explicit suffix
  --> $DIR/numeric_literal_missing_suffix.rs:17:18
   |
LL |     let x: u8 = 1;
   |                  ^ help: add it: `u8`

error: this literal is missing an explicit suffix
  --> $DIR/numeric_literal_missing_suffix.rs:18:23
   |
LL |     static X1: u32 = 0;
   |                       ^ help: add it: `u32`

error: this literal is missing an explicit suffix
  --> $DIR/numeric_literal_missing_suffix.rs:19:22
   |
LL |     const X2: u32 = 0;
   |                      ^ help: add it: `u32`

error: aborting due to 5 previous errors

